text_models/bert.py

```diff
@@ -29,25 +30,56 @@ def __len__(self):
         return len(self.encodings.input_ids)
 
 
+class BertModelMLMDataset(BertModelDataset):
+    def __init__(self, encodings, mask_id=103, cls_id=102, sep_id=101, pad_id=0, mask_probability=0.15):
+        super(BertModelMLMDataset, self).__init__(encodings)
+
+        self.encodings["labels"] = self.encodings.input_ids.detach().clone()
+        self.mask_proba = mask_probability
+        self.mask_id = mask_id
+        self.cls_id = cls_id
+        self.sep_id = sep_id
+        self.pad_id = pad_id
+
+    def __getitem__(self, idx):
+        inputs = self.encodings.input_ids[idx]
+
+        rand = torch.rand(inputs.shape)
+        mask_arr = (
+            (rand < self.mask_proba) * (inputs != self.cls_id) * (inputs != self.sep_id) * (inputs != self.pad_id)
+        )
+        inputs[mask_arr] = self.mask_id
```

Comment:
If I understand correctly, every time you query the same element, it becomes more and more "masked". It should not be the case.

Reply:
Fixed